morpheus/_lib/src/io/loaders/rest.cpp

```diff
@@ -17,21 +17,378 @@
 
```

Comment:
One high level comment:

If we have functions that are only ever intended to be called within this `rest.cpp` compilation unit, please put them inside an anonymous namespace. Otherwise, while they're not part of the public API, their symbols are still accessible, and it is also possible to violate ODR within the project in a non-obvious way.

See my comment on the `convert_url_query_to_dict` function.